lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -866,5 +866,7 @@ void populateElementwiseOpToLLVMPatterns(mlir::LLVMTypeConverter &typeConverter,
   // For FP64 input type, ExpOpConversionApprox will return failure and
   // ElementwiseOpConversion<math::ExpOp, math::ExpOp> defined below will call
   // __nv_expf for higher-precision calculation
+#ifndef USE_ROCM
```

Comment:
@rsanthanam-amd lplease provide comments to explain why this is not needed in rocm path.

Reply:
The ExpOpConversionApprox construct is a PTX specific implementation detail that most likely offers higher performance for fp32 but does not work for fp64.

By not using this for AMDGPU, we ensure that exp works for all float data types.

This leverages the cuda2gcn wrapper along with https://github.com/ROCmSoftwarePlatform/triton/pull/68 to enable exp for AMDGPU.